src/hdtXmlPatternExpander.cpp

```
@@ -0,0 +1,722 @@
+			for (std::size_t j = i; j < s.size(); ++j)
+				if (!std::isdigit(static_cast<unsigned char>(s[j])))
+					return std::nullopt;
+			try {
+				return std::stol(s);
+			} catch (...) {
+				return std::nullopt;
+			}
+		}
+
+		// 1-based line number of a byte offset in the original text (0 if the offset is unknown).
+		int offsetToLine(const std::string& s, std::ptrdiff_t off)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)